validator_client/src/beacon_node_fallback.rs

```diff
@@ -420,26 +467,52 @@ impl<T: SlotClock, E: EthSpec> BeaconNodeFallback<T, E> {
     /// low quality responses. To route around this it's best to poll all connected beacon nodes.
     /// A previous implementation of this function polled only the unavailable BNs.
     pub async fn update_all_candidates(&self) {
-        let futures = self
-            .candidates
-            .iter()
-            .map(|candidate| {
-                candidate.refresh_status(self.slot_clock.as_ref(), &self.spec, &self.log)
-            })
-            .collect::<Vec<_>>();
+        let candidates = self.candidates.read().await;
```

Comment:
Oh I missed this one - i think this one needs to be fixed, because we only drop the read lock after refresh health (api calls to all BNs) is completed

Reply:
Pushed fix here https://github.com/sigp/lighthouse/pull/4393/commits/30ddfa600d31a8041192488d7a9101a122cd809b